packages/protocol/contracts/layer1/shasta/impl/Inbox.sol

```diff
@@ -149,19 +146,21 @@ abstract contract Inbox is EssentialContract, IInbox {
 
         // Verify capacity for new proposals
         uint256 availableCapacity = _getAvailableCapacity(config, coreState);
-        require(availableCapacity > 0, ExceedsUnfinalizedProposalCapacity());
+        require(availableCapacity > input.numForcedInclusions, ExceedsUnfinalizedProposalCapacity());
 
         // Process forced inclusion if required
-        bool forcedInclusionProcessed;
-        (coreState, forcedInclusionProcessed) = _processForcedInclusion(config, coreState);
-
-        if (!forcedInclusionProcessed || availableCapacity > 1) {
-            // Propose the normal proposal after the potential forced inclusion to match the
-            // behavior in Shasta fork.
-            LibBlobs.BlobSlice memory blobSlice =
-                LibBlobs.validateBlobReference(input.blobReference, _getBlobHash);
-            _propose(config, coreState, blobSlice, false);
+        coreState = _processForcedInclusions(config, coreState, input.numForcedInclusions);
+
+        // Verify that no forced inclusion that is due remains in the queue.
+        if (IForcedInclusionStore(config.forcedInclusionStore).isOldestForcedInclusionDue()) {
```

Comment:
use require-statement for consistency?

Reply:
There is a potential issue here, an attacker can create many forced inclusion with the same timestamp, so they will due all together, even if the current proposer have consumed many inclusions, the next one is still due. 

Therefore, I think we can change the if-statement to

```
require(input.numForcedInclusions  > 1 || 
!IForcedInclusionStore(config.forcedInclusionStore).isOldestForcedInclusionDue(), 
UnprocessedForcedInclusionIsDue());
```